src/v/kafka/protocol/schemata/generator.py

```diff
@@ -1184,7 +1221,7 @@ class response;
 {%- set decoder, named_type = field.decoder(flex) %}
 {%- if named_type == None %}
     return reader.{{ decoder }};
-{%- elif field.nullable() %}
+{%- elif field.nullable() and field.has_primitives == false %}
```

Comment:
@dotnwat This is the workaround I came up with for the issue with our template generator.

To recap, the AlterPartitionReassignmentsRequest Schemata has the following form for the replica set of a partition:
```json
       { "name": "Replicas", "type": "[]int32", "versions": "0+", "nullableVersions": "0+", "default": "null", "entityType": "brokerId",
          "about": "The replicas to place the partitions on, or null to cancel a pending reassignment for this partition." }
```

The issue is that `nullableVersions` and `entityType` are set and out generator thinks that the elements within the Replicas vector are **also optional**.  This leads to a compile error such as:
```
[2/63] Building CXX object src/v/kafka/protocol/CMakeFiles/v_kafka_protocol.dir/schemata/protocol.cc.o
FAILED: src/v/kafka/protocol/CMakeFiles/v_kafka_protocol.dir/schemata/protocol.cc.o 
/usr/bin/ccache /home/nyalia/redpanda/vbuild/llvm/install/bin/clang++ -DBASE64_STATIC_DEFINE -DBOOST_ASIO_HAS_STD_INVOKE_RESULT -DFMT_LOCALE -DSEASTAR_API_LEVEL=6 -DSEASTAR_DEBUG -DSEASTAR_DEBUG_SHARED_PTR -DSEASTAR_DEFAULT_ALLOCATOR -DSEASTAR_SCHEDULING_GROUPS_COUNT=16 -DSEASTAR_SHUFFLE_TASK_QUEUE -DSEASTAR_TYPE_ERASE_MORE -DXXH_PRIVATE_API -DZSTD_STATIC_LINKING_ONLY -I/home/nyalia/redpanda/src/v -I/home/nyalia/redpanda/vbuild/debug/clang/src/v -isystem /home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include -isystem /home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include/hdr -fPIC -march=westmere -fuse-ld=lld -mllvm -inline-threshold=2500 -Wno-unused-lambda-capture -Wno-unused-command-line-argument -stdlib=libc++ -fsanitize=undefined -fsanitize=address -O0 -g -gdwarf-4 -gsplit-dwarf -fPIC -Wall -Wextra -Werror -Wno-missing-field-initializers -U_FORTIFY_SOURCE -DSEASTAR_SSTRING -Werror=unused-result "-Wno-error=#warnings" -fstack-clash-protection -fsanitize=address -fsanitize=undefined -fno-sanitize=vptr -std=c++20 -MD -MT src/v/kafka/protocol/CMakeFiles/v_kafka_protocol.dir/schemata/protocol.cc.o -MF src/v/kafka/protocol/CMakeFiles/v_kafka_protocol.dir/schemata/protocol.cc.o.d -o src/v/kafka/protocol/CMakeFiles/v_kafka_protocol.dir/schemata/protocol.cc.o -c /home/nyalia/redpanda/vbuild/debug/clang/src/v/kafka/protocol/schemata/protocol.cc
/home/nyalia/redpanda/vbuild/debug/clang/src/v/kafka/protocol/schemata/protocol.cc:7761:57: error: indirection requires pointer operand ('int32_t' (aka 'int') invalid)
                        return model::node_id(std::move(*tmp));
                                                        ^~~~
/home/nyalia/redpanda/vbuild/debug/clang/src/v/kafka/protocol/schemata/protocol.cc:7756:33: error: no matching member function for call to 'read_nullable_flex_array'
            v.replicas = reader.read_nullable_flex_array([version](request_reader& reader) {
                         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/home/nyalia/redpanda/src/v/kafka/protocol/request_reader.h:178:5: note: candidate template ignored: substitution failure [with ElementParser = (lambda at /home/nyalia/redpanda/vbuild/debug/clang/src/v/kafka/protocol/schemata/protocol.cc:7756:58)]: no type named 'type' in 'std::invoke_result<(lambda at /home/nyalia/redpanda/vbuild/debug/clang/src/v/kafka/protocol/schemata/protocol.cc:7756:58), kafka::request_reader &>'
    read_nullable_flex_array(ElementParser&& parser) {
    ^
2 errors generated.
```

Reply:
The workaround here is to specifically check for an array of primitive types.

Reply:
Looks like the deletion will work.